src/lib/tvgSceneImpl.h

```diff
@@ -81,8 +86,14 @@ struct Scene::Impl
 
     bool needComposition(uint32_t opacity)
     {
+        if (opacity == 0 || paints.count == 0) return false;
+
+        //Masking may require composition (even if opacity == 255)
+        auto compMethod = scene->composite(nullptr);
+        if (compMethod != CompositeMethod::None && compMethod != CompositeMethod::ClipPath) return true;
+
         //Half translucent requires intermediate composition.
-        if (opacity == 255 || opacity == 0) return false;
+        if (opacity == 255) return false;
 
         //If scene has several children or only scene, it may require composition.
```

Comment:
these can be replaced by:
```
    if (paints.count == 1 && (*paints.data)->identifier() != TVG_CLASS_ID_SCENE) return false;
    return true;
```
but I think it's less readable so I kept this as it was.